added to catalog layout for consistent levels (#1009)
radiantearth/stac-spec: best-practices.md

5. Items should be stored in subdirectories of their parent catalog. 
This means that each item and its assets are contained in a unique subdirectory.
6. Limit the number of items in a catalog or sub-catalog, grouping / partitioning as relevant to the dataset.

> 7. In a nested hierarchy of catalogs and collections make the different levels 'homogeneous' - all catalogs or all collections. 

Reviewer: That sounds a bit like there should be only one level of collections, although it should also be fine to have e.g. all levels being collections.

Author: Cool, I'll try again. I was thinking about that as I was writing it but didn't manage to find a way to succinctly get it across. This was weirdly more challenging to explain than I was expecting.

5. Items should be stored in subdirectories of their parent Catalog. 
This means that each Item and its assets are contained in a unique subdirectory.
6. Limit the number of Items in a Catalog or sub-Catalog, grouping / partitioning as relevant to the dataset.

> 7. Use structural elements (Catalog and Collection) consistently across each 'level' of your hierarchy. If levels 2 and 4
> have are both only Collections then don't add a Catalog there (and vice versa), or else change every element at that level.

Reviewer 2: ```suggestion
7. Use structural elements (Catalog and Collection) consistently across each 'level' of your hierarchy. For example, if levels 2 and 4 of the hierarchy only contain Collections, 
don't add a Catalog at levels 2 and 4.
```
